operations/patch_util.go

```diff
@@ -368,100 +368,110 @@ func formatRev(rev string) string {
 	return fmt.Sprintf("%s^!", rev)
 }
 
+func getFeatureBranch(ref, commits string) string {
+	if ref != "" {
+		return ref
+	}
+	if commits != "" {
+		// if one commit, this returns just that commit, else the first commit in the range
+		return strings.Split(commits, "..")[0]
+	}
+	return "HEAD"
+}
+
 // loadGitData inspects the current git working directory and returns a patch and its summary.
 // The branch argument is used to determine where to generate the merge base from, and any extra
 // arguments supplied are passed directly in as additional args to git diff.
-func loadGitData(branch, ref, rev string, format bool, extraArgs ...string) (*localDiff, error) {
+func loadGitData(branch, ref, commits string, format bool, extraArgs ...string) (*localDiff, error) {
 	// branch@{upstream} refers to the branch that the branch specified by branchname is set to
 	// build on top of. This allows automatically detecting a branch based on the correct remote,
-	// if the user's repo is a fork, for example.
+	// if the user's repo is a fork, for example. This also works with a commit hash, if given.
+	// In the case a range is passed, we only need one commit to determine the base, so we use the last commit.
```

Comment:
it looks like we're using the first commit

Reply:
Yes sorry I changed this later.